lib/internal/bootstrap/node.js

```diff
@@ -625,5 +627,59 @@
     new vm.Script(source, { displayErrors: true, filename });
   }
 
+  function setupAllowedFlags() {
+    // This builds process.allowedNodeEnvironmentFlags
+    // from data in the config binding
+
+    const replaceDashesRegex = /-/g;
+    const leadingDashesRegex = /^--?/;
+    const trailingValuesRegex = /=.*$/;
+
+    const trimLeadingDashes = (flag) => flag.replace(leadingDashesRegex, '');
+
+    const {
+      allowedV8EnvironmentFlags,
+      allowedNodeEnvironmentFlags
+    } = process.binding('config');
+
+    // for comparison purposes only
+    const v8Flags = new Set(allowedV8EnvironmentFlags
+      .map(trimLeadingDashes));
+    const nodeFlags = new Set(allowedNodeEnvironmentFlags
```

Comment:
Id suggest on both of these sets, doing `Object.defineProperties(set, Object.getOwnPropertyDescriptors(Set.prototype))` to be robust against later mutations of Set.prototype

Reply:
ok, will do. 

at the same time, what exactly are we preventing by locking down this object?  I have to wonder given the extra effort to do so.

Reply:
It's preventing something in node core breaking because of arbitrary user code.